lib/features/log/Log.js

```diff
@@ -151,96 +198,142 @@ export default function Log(
       this.log({
-        text: elementName || 'Script Task',
-        icon: 'bpmn-icon-script',
+        text: elementName || 'Receive Task',
+        icon: 'bpmn-icon-receive',
         scope
       });
-    } else if (is(element, 'bpmn:ServiceTask')) {
+      return;
+    }
+    if (is(element, 'bpmn:SendTask')) {
       this.log({
-        text: elementName || 'Service Task',
-        icon: 'bpmn-icon-service',
+        text: elementName || 'Send Task',
+        icon: 'bpmn-icon-send',
         scope
       });
-    } else if (is(element, 'bpmn:Task')) {
+      return;
+    }
+    if (is(element, 'bpmn:Task')) {
       this.log({
         text: elementName || 'Task',
         icon: 'bpmn-icon-task',
         scope
       });
-    } else if (is(element, 'bpmn:UserTask')) {
-      this.log({
-        text: elementName || 'User Task',
-        icon: 'bpmn-icon-user',
-        scope
-      });
-    } else if (is(element, 'bpmn:ExclusiveGateway')) {
+      return;
+    }
+
+    // Log gateways
+    if (is(element, 'bpmn:ExclusiveGateway')) {
       if (element.outgoing.length < 2) {
         return;
       }
 
+      // TODO: The property sequenceFlow does not exist. Logging the name is currently not working.
       const sequenceFlowName = getElementName(element.sequenceFlow);
 
       let text = elementName || 'Gateway';
 
       if (sequenceFlowName) {
-        text = text.concat(` ${ AngleRightIcon() } ${ sequenceFlowName }`);
+        text = text.concat(` ${AngleRightIcon()} ${sequenceFlowName}`);
       }
 
       this.log({
         text,
         icon: 'bpmn-icon-gateway-xor',
         scope
       });
-    } else if (is(element, 'bpmn:EndEvent')) {
-      if (isTypedEvent(getBusinessObject(element), 'bpmn:TerminateEventDefinition')) {
-        this.log({
-          text: elementName || 'Terminate End Event',
-          icon: 'bpmn-icon-end-event-terminate',
-          scope
-        });
-      } else {
-        this.log({
-          text: elementName || 'End Event',
-          icon: 'bpmn-icon-end-event-none',
-          scope
-        });
-      }
-    } else if (is(element, 'bpmn:StartEvent')) {
+      return;
+    }
+
+    // Log events
+    if (is(element, 'bpmn:StartEvent')) {
       this.log({
         text: elementName || 'Start Event',
-        icon: 'bpmn-icon-start-event-none',
+        icon: `bpmn-icon-start-event-${getEventTypeString(element)}`,
+        scope
+      });
+      return;
+    }
+    if (is(element, 'bpmn:IntermediateCatchEvent')) {
+      this.log({
+        text: elementName || 'Intermediate Event',
+        icon: getIconForIntermediateEvent(element, 'catch'),
+        scope
+      });
+      return;
+    }
+    if (is(element, 'bpmn:IntermediateThrowEvent')) {
+      this.log({
+        text: elementName || 'Intermediate Event',
+        icon: getIconForIntermediateEvent(element, 'throw'),
+        scope
+      });
+      return;
+    }
+    if (is(element, 'bpmn:BoundaryEvent')) {
+      this.log({

```

Comment:
```suggestion
      return this.log({

```

If we apply an eager return instead of `if/else` nesting, let's do it like this :arrow_up:

Reply:
We can do that, but why do you prefer that?
We should then probably do it consistently for all cases.

Reply:
Yes of course. If we drop nesting, let's establish

```
return iDidWork()
```

as a pattern.

Reply:
Just changed it.